[DOC] Getting Started with OpenSearch Dashboards (#1009)
opensearch-project/documentation-website: _dashboards/index.md

---
layout: default
title: Getting Started with OpenSearch Dashboards

> nav_order: 1

Reviewer: Could we move the Getting started page to before the Installations page?

{%- comment -%}The `/docs/opensearch-dashboards/` redirect is specifically to support the UI links in OpenSearch Dashboards 1.0.0.{%- endcomment -%}

# What is OpenSearch Dashboards?

> OpenSearch Dashboards is an open-source analytics and search application for the OpenSearch engine. With OpenSearch Dashboards, you can query, visualize, alert on, and understand your metrics no matter where they are stored. 

Reviewer 2: The "no matter where they are stored" may be misleading. At the moment, the data will need to be stored in their OpenSearch cluster. In the future, this will be different.

Author: Rewrote section

{%- comment -%}The `/docs/opensearch-dashboards/` redirect is specifically to support the UI links in OpenSearch Dashboards 1.0.0.{%- endcomment -%}

# What is OpenSearch Dashboards?

> OpenSearch Dashboards is an open-source analytics and search application that allows you to monitor events, make decisions, inform others, and identify trends. It displays multiple visualizations that work together on a single screen, giving you a centralized and comprehensive view of your data and providing key insights for at-a-glance decision-making.

Reviewer 3: Why do I feel like we have lost more information with the new description? Most of the features mentioned like monitoring events, informing others, identifying trends are actually a feature provided by plugins that are installed over the base installation of Dashboards. The basic installation does not have any of those features. 

The Base installation comes with 3 main apps:
- Discover: The ability to see the OpenSearch data
- Visualize: create visualizations for the data
- Dashboard: display the visualizations on a dashboard

Apart from these basic apps, Dashboards also provides tools to interact with OpenSearch and a plugin architecture and ecosystem to extend its basic functions. e.g. Observability, alerting, anomaly detection, security etc

Reviewer 4: I think that's one of the big problems and limitations with our naming - "OpenSearch Dashboards" can have different meanings in different contexts. Unless you're installing the minimal OSD version (which seems relatively rare, anecdotally), you will get the full suite of plugins/features mentioned above by default. We don't currently have a good way of distinguishing between OpenSearch Dashboards as an extensible platform with some minimal features, and OpenSearch Dashboards as experienced by most end users.

Reviewer 3: We can call that out though right?  Based on this, a user who installs the minimal distribution would not have any of the features listed above and be caught off guard. We can use this doc to separate the two, but calling dashboards a tool for alerting, anomaly detection when it's not is kinda limiting and misleading. Infact if anything the fact that a plugin can extend to do all that should be highlighted as the power of dashboards to extend and support such complex features and more!

> OpenSearch Dashboards is an open-source analytics and search application that gives you a view of information and data all in one pane. OpenSearch Dashboards allows you to monitor events, make decisions, inform others, and see trends and displays multiple visualizations that work together on a single screen. These visuals give you a comprehensive view of your data and provide key insights for at-a-glance decision-making.

Author: Accepted suggestions. Clear and concise.

![Visualization example image](../images/what-is-dashboards.jpg)

> OpenSearch Dashboards allows you to create dashboards for many reasons, and many types of dashboards. OpenSearch Dashboards lets you to do the following:  

Reviewer 5: ```suggestion
OpenSearch Dashboards allows you to create several types of dashboards for many different purposes. Some of the ways you can use Dashboards include:   
```

> * See-all in one view-the data you need to make informed decisions.

Reviewer 6: ```suggestion
* See the data you need to make informed decisions in a single pane.
```

> * See-all in one view-the data you need to make informed decisions.

Reviewer 6: ```suggestion
* See an all-in-one view of the data you need to make informed decisions.
```

Reviewer 6: just a note, pls make commit separately from Github (don't click "Commit suggestion" button), it will create a non-valid DCO signature.

* Search, explore, filter, aggregate, and visualize your data in near real time.
* Track and monitor the most important information about your day-to-day operations. 
* Schedule, export, and share reports from dashboards, saved searches, alerts, and visualizations. 

> The type of dashboards you create depends on the  different teams and people within and outside your organization. OpenSearch Dashboards are created to be intuitive and easy to understand by users like business analysts, senior executives, engineering managers, and operations managers.  

Reviewer 5: ...type of dashboard? or ...types of dashboards....?

Author: types

> The types of dashboards you create depends on the  different teams and people within and outside your organization. OpenSearch Dashboards are created to be intuitive and easy to understand by users like business analysts, senior executives, engineering managers, and operations managers.  

Author: Agreed. Revised.

{%- comment -%}The `/docs/opensearch-dashboards/` redirect is specifically to support the UI links in OpenSearch Dashboards 1.0.0.{%- endcomment -%}

# Getting started with OpenSearch Dashboards

OpenSearch Dashboards is the preferred visualization platform for analyzing data in OpenSearch. It serves as the user interface for many openSearch features, including security, alerting, index state management, SQL, and more.

> Out-of-the-box OpenSearch Dashboards includes several tools for discovering and visualizing data, and OpenSearch provides a full suite of plugins that you can use to monitor events, identify trends, and automate recurring activities.

Reviewer 5: Do we need to differentiate that OpenSearch Dashboards has its own plugins? We should probably link to the topic in the related links.

Author: Added links under related topics

Reviewer 4: I think it depends on the audience for this doc and which usage of "OpenSearch" we're thinking of here. The OpenSearch project/organization does provide a full suite of plugins; some of these are OpenSearch Dashboards plugins (front-end only), some are OpenSearch plugins (backend only), and some that require both types of plugins installed together to work correctly (e.g. Security). That feels like a whole can of technical implementation worms to open here if the intended audience won't care.

> ![Sample dashboards](../images/dashboards-example.png)

Reviewer 5: For accessibility, we need to describe the image. At the least, introduce the image and what we are looking at.

Author: Added alt text and changed URL

# Try OpenSearch Dashboards
You can try OpenSearch Dashboards' features in the [OpenSearch Dashboards Playground](https://playground.opensearch.org/app/home).

Choose **OpenSearch Dashboards > Dashboards > Create dashboard** and explore the sample Global Flight, Web Traffic, and Revenue dashboards.

> 

Reviewer 5: Again, we need to introduce and describe what we are showing in the image.

> ![Dashboards playground screenshot](../images/dashboards-playground.png)

Reviewer: You cannot use relative paths in links to our site. Use full path with a dynamic site url, as Jeff suggested.

Reviewer: I'm questioning if this image adds any information. On the one hand, it's a visual, so it can draw attention. On the other hand, this is exactly what they will see if they go to the playground link... @hdhalter - could you weigh in?

# OpenSearch Dashboards for your use case

OpenSearch Dashboards makes it easy to visualize data for a number of use cases such as log analytics, application search, enterprise search, and more. Here are a few examples.  

> <table>

Reviewer: Use this table instead. It will center the text and images within cells:
<table>
<tbody>
<tr>
<td style="text-align: center; vertical-align: top;">Visualize log and trace data with interactive log analytics<img src="{{site.url}}{{site.baseurl}}/images/trace_log_correlation-web.jpg" alt="Log analytics"/></td>
<td style="text-align: center; vertical-align: top;">Centralize and analyze logs for real-time threat detection and incident management<img src="{{site.url}}{{site.baseurl}}//images/detection.png" alt="Anomaly detection"/></td>
</tr>
<tr>
<td style="text-align: center; vertical-align: top;">Quickly find relvant data within your applications, websites, and data catalogs<img src="{{site.url}}{{site.baseurl}}/images/drag-and-drop.png" alt="Data overview" /></td>
<td style="text-align: center; vertical-align: top;">Find and fix problems and improve application health at a glance<img src="{{site.url}}{{site.baseurl}}/images/dashboards5.PNG" alt="Event analysis" /></td>
</tr>
</tbody>
</table>

<thead>
</thead>
<tbody>
<tr>
<td>Visualize log and trace data with interactive log analytics<img src="{{site.url}}{{site.baseurl}}/images/trace_log_correlation-web.jpg" alt="Log analytics" width="400" style="float: left; margin-right: 15px;"/></td>

> <td>Centralize and analyze logs for real-time threat detection and incident management<img src="{{site.url}}{{site.baseurl}}//images/detection.png" alt="Anomaly detection" width="400" style="float: left; margin-right: 15px;"/></td>

Reviewer: Could we clean up this image? It has extra stuff on the right, so it's not going to center properly. Also, the very bottom of it can be cropped.

</tr>
<tr>

> <td>Quickly find relvant data within your applications, websites, and data catalogs<img src="{{site.url}}{{site.baseurl}}/images/drag-and-drop.png" alt="Data overview" width="400" style="float: left; margin-right: 15px;"/></td>

Reviewer: Relevant

Reviewer: What is this referring to? It's not clear to me what this use case is...

> <td>Find and fix problems and improve application health at a glance<img src="{{site.url}}{{site.baseurl}}/images/dashboards5.PNG" alt="Event analysis" width="400" style="float: left; margin-right: 15px;"/></td>

Reviewer: This is not the right image

Author: Replaced with correct image.

Reviewer: Could this be reworded? It reads as if we're improving application health at a glance 😄

</tr>
</tbody>
</table>

> # Install and create OpenSearch Dashboards

Reviewer: Can we rephrase this title? Technically, they are not creating a dashboard. They're just trying out Dashboards locally by pulling up a dashboard that has been created for them.
So, something like "Install and try out OpenSearch Dashboards locally"

> After installing OpenSearch, you can install OpenSearch Dashboards, with different installation methods to choose. For more information, see [OpenSearch Installation Instructions](https://opensearch.org/docs/latest/opensearch/install/index/).

Reviewer 5: Shouldn't we point to the OpenSearch Dashboards installation topic?

> You can follow these steps to create a dashboard using sample data. 

Reviewer 5: Shouldn't these steps be combined with 'Try OpenSearch Dashboards'?

1. After installing OpenSearch Dashboards, open it at [http://localhost:5601](http://localhost:5601/app/home#/).
2. Select **Add sample data**, and select **Add data** for the sample  flight data dataset. 
3. From the top menu, choose **Discover** and search for a few flights.
4. Choose **Dashboard**, **[Flights]** Global Flight Dashboard.

> Your dashboard will look different than the example below.

Reviewer: Is there any reason it would look different? If all they did was add the flight dataset and pull up a pre-made Flights dashboard, it should look exactly the same. Or is the picture not reflecting what they should see?

> ![Sample dashboard using flight sample data](../images/flight-dashboards.png)

Reviewer: Again, change the relative path in the link as above.

{%- comment -%}The `/docs/opensearch-dashboards/` redirect is specifically to support the UI links in OpenSearch Dashboards 1.0.0.{%- endcomment -%}

# Overview of OpenSearch Dashboards

> OpenSearch Dashboards is the preferred visualization platform for analyzing data in OpenSearch. and is the user interface for many OpenSearch features. From real-time application monitoring, threat detection, and incident management to personalized search experience, OpenSearch Dashboards gives you data visualization tools to tell your data story.      

Reviewer 4: ```suggestion
OpenSearch Dashboards is the preferred visualization platform for analyzing data in OpenSearch, and is the user interface for many OpenSearch features. From real-time application monitoring, threat detection, and incident management to personalized search experience, OpenSearch Dashboards gives you data visualization tools to tell your data story.      
```

OpenSearch Dashboards is the default visualization tool for data in OpenSearch. It also serves as a user interface for many of the OpenSearch plugins, including security, alerting, Index State Management, SQL, and more.
OpenSearch Dashboards is the preferred visualization platform for analyzing data in OpenSearch. and is the user interface for many OpenSearch features. From real-time application monitoring, threat detection, and incident management to personalized search experience, OpenSearch Dashboards gives you data visualization tools to tell your data story.      

Out-of-the-box OpenSearch Dashboards includes several tools for discovering and visualizing data, and the OpenSearch suite provides [plugins](https://opensearch.org/docs/latest/dashboards/install/plugins/) to extend and customize dashboard features and functions. 

> ## Get started with OpenSearch Dashboards

Reviewer 4: Is this still intended to be a subhead?

OpenSearch Dashboards makes it easy makes it easy for you to perform interactive log analytics, real-time application monitoring, website search, and more.  

<table>
<tbody>
<tr>

> <td style="text-align: center; vertical-align: top;">Visualize log and trace data with interactive log analytics<img src="{{site.url}}{{site.baseurl}}/images/trace_log_correlation.gif)" alt="Log analytics"/></td>

Reviewer 7: Should each of the sentences end in a period?

<td style="text-align: center; vertical-align: top;">Detect and mitigate issues faster with anomaly detection<img src="{{site.url}}{{site.baseurl}}/images/anomaly-detection.png" alt="Anomaly detection"/></td>
</tr>
<tr>

> <td style="text-align: center; vertical-align: top;">Monitor and issue alerts with observability<img src="{{site.url}}{{site.baseurl}}/images/observability.png" alt="Data overview" /></td>

Reviewer 7: Confirm whether "observability" should be capitalized.

> <td style="text-align: center; vertical-align: top;">Use observability to monitor and issue alerts<img src="{{site.url}}{{site.baseurl}}/images/observability.png" alt="Data overview" /></td>

Reviewer 4: If we're following the "task/feature" pattern of the first two items, it would be:

```suggestion
<td style="text-align: center; vertical-align: top;">Monitor and issue alerts with observability<img src="{{site.url}}{{site.baseurl}}/images/observability.png" alt="Data overview" /></td>
```

<td style="text-align: center; vertical-align: top;">Unlock real-time search, monitoring, and analysis of business and operational data<img src="{{site.url}}{{site.baseurl}}/images/real-time-insights.png" alt="Real-time search, monitoring, and analysis" /></td>
</tr>
</tbody>
</table> 

Follow the steps below to get started creating a dashboard and exploring its elements.

> # Try out OpenSearch Dashboards on the playground

Reviewer 7: Will we have explained what the playground is prior to this? If not, this will be unclear to the reader. Confirm whether this should be "Try out OpenSearch Dashboards on the OpenSearch Dashboards Playground".

Follow these steps to learn the basics of OpenSearch Dashboards:

1.  Go to the [OpenSearch Dashboards Playground](https://playground.opensearch.org/app/home).
2.  Choose **OpenSearch Dashboards > Dashboards > Create dashboard**. 
3.  Explore the sample Global Flight, Web Traffic, and Revenue dashboards. 

# Install OpenSearch Dashboards

You can try out OpenSearch Dashboards locally after installing OpenSearch. For installation instructions, see [Install and configure OpenSearch](https://opensearch.org/docs/latest/opensearch/install/index/) and [Install and configure OpenSearch Dashboards](https://opensearch.org/docs/latest/dashboards/install/index/). 

> # Try out OpenSearch Dashboards locally 

Reviewer 4: There's some redundancy here - I think either both headers should include "locally" or neither should, and the fact that installation is a per-requisite seems obvious enough that we can probably skip that call-out.

Follow these steps to create a dashboard on localhost using sample data: 

1. After installing OpenSearch Dashboards, access it on [localhost:5601](http://localhost:5601/app/home#/).

> 2. Select **Add sample data**, and select **Add data** for the sample  flight data dataset. 

Reviewer 7: It looks like there's an extra space between "sample" and "flight" here, but it doesn't appear under "View file", so it may not be an issue.

3. From the top menu, choose **Discover** and search for a few flights.

> 4. Choose **Dashboard**, **[Flights]** Global Flight Dashboard.

Reviewer 7: Should "Global Flight Dashboard" be included in the bold text?

# What is OpenSearch Dashboards?

OpenSearch Dashboards is the preferred visualization platform for analyzing data in OpenSearch. It serves as the user interface for many openSearch features, including security, alerting, index state management, SQL, and more.

Out-of-the-box OpenSearch Dashboards includes several tools for discovering and visualizing data, and OpenSearch provides a full suite of plugins that you can use to monitor events, identify trends, and automate recurring activities.

> ![Sample dashboards]({site.url}}{{site.baseurl}}/images/dashboards-example.png)

Reviewer 8: Missed a curly bracket at the beginning of `site.url` - probably just a copy-paste error but make sure to add it otherwise the link won't display correctly.

This looks like it's the original page though but still needs fixed before publishing. Does this link work when you try to build it locally?

<td style="text-align: center; vertical-align: top;">Visualize log and trace data with interactive log analytics<img src="{{site.url}}{{site.baseurl}}/images/trace_log_correlation-web.jpg" alt="Log analytics"/></td>
<td style="text-align: center; vertical-align: top;">Detect and mitigate issues faster with anomaly detection<img src="{{site.url}}{{site.baseurl}}/images/anomaly-detection.png" alt="Anomaly detection"/></td>
</tr>
<tr>
<td style="text-align: center; vertical-align: top;">Use observability to monitor and issue alerts<img src="{{site.url}}{{site.baseurl}}/images/observability.png" alt="Data overview" /></td>
<td style="text-align: center; vertical-align: top;">Unlock real-time search, monitoring, and analysis of business and operational data<img src="{{site.url}}{{site.baseurl}}/images/real-time-insights.png" alt="Real-time search, monitoring, and analysis" /></td>
</tr>
</tbody>
</table>

# Install and try OpenSearch Dashboards

After installing OpenSearch, you can install OpenSearch Dashboards, with different installation methods to choose. For more information, see [OpenSearch Installation Instructions](https://opensearch.org/docs/latest/opensearch/install/index/).

> You can try OpenSearch Dashboards' features in the [OpenSearch Dashboards Playground](https://playground.opensearch.org/app/home). Choose **OpenSearch Dashboards > Dashboards > Create dashboard** and explore the sample Global Flight, Web Traffic, and Revenue dashboards. You can follow these steps to create a dashboard using sample data. 

Reviewer: This should be two separate sections (like before):
1. Try dashboards with playground
 Choose **OpenSearch Dashboards > Dashboard**. Choose a dashboard from the existing ones and explore.
3. Try dashboards after installing it locally
(the instructions you have)
Please separate the instructions into these two categories.

1. After installing OpenSearch Dashboards, open it at [http://localhost:5601](http://localhost:5601/app/home#/).
2. Select **Add sample data**, and select **Add data** for the sample  flight data dataset. 
3. From the top menu, choose **Discover** and search for a few flights.
4. Choose **Dashboard**, **[Flights]** Global Flight Dashboard.

> Here's an example of the dashboard you created.

Reviewer: You can  just say "You should see the following dashboard" or something like that. They have not created a dashboard, they just opened one that comes with the flights data. Otherwise, the users might be confused.

Choose **OpenSearch Dashboards > Dashboards > Create dashboard** and explore the sample Global Flight, Web Traffic, and Revenue dashboards.

> ![Dashboards playground screenshot]({site.url}}{{site.baseurl}}/images/dashboards-playground.png)

Reviewer 8: Missed a curly bracket at the beginning of `site.url` - probably just a copy-paste error but make sure to add it otherwise the link won't display correctly.

# Install OpenSearch Dashboards

> After installing OpenSearch, you can install OpenSearch Dashboards, with different installation methods to choose from. For more information, see [OpenSearch Installation Instructions](https://opensearch.org/docs/latest/opensearch/install/index/).

Reviewer 8: The name of the link should be `Install and configure OpenSearch` so it aligns with the header on the page. You also need to make sure that the base URL is dynamic, since you don't know which version the user is browsing. By hard-coding in a URL with `/latest/` in the path you would force all users to the newest version supported by the website.

Change the link part to:

[Install and configure OpenSearch]({{site.url}}{{site.baseurl}}/opensearch/install/index/)

3. From the top menu, choose **Discover** and search for a few flights.
4. Choose **Dashboard**, **[Flights]** Global Flight Dashboard.

> You have now created a dashboard, which may look different than the example below.

Reviewer: Technically, they have not **created** the dashboard, they just pulled up the sample one that is provided. So, no need for this sentence at all.

Author: Revised
